### DISCUSSION TITLE
When i set value default. Component re-render but value default not change. How to change it? Thanks

### DISCUSSION THREAD

**Discussion by megabeehothanhminh** [MobileReality/react-native-select-pro#75] [Q&A]

**Comment by irekrog** [comment]
Hi @megabeehothanhminh 

You can't do this. This is the correct behavior. [Look at the examples from two popular select libraries](https://codesandbox.io/s/thirsty-cloud-5knkns?file=/src/App.js) (react-select and mantine). After click a button a component is re-render but visible default value is still the same.

If you want then please share your code or describe the broader context of your problem then we can help you.

**Reply by megabeehothanhminh** [reply]
Okee thx. I will look at other ways

**Comment by gbspecapedra** [comment]
In my project a generic select was created and when I set the defaultOption I went through the same problem, and to get around this I did the following using react-hook-form:

```
const { getValues } = useFormContext();

const [defaultOption, setDefaultOption] = useState<OptionTypeRequired | undefined>(
    options.find((item) => item.value === getValues(name))
  );

<Controller
    name={name}
    options={options}
    render={() => (
        <Select
          options={options}
          defaultOption={defaultOption}
          onRemove={() => setDefaultOption(undefined)}
        />
    )}
 />
```
